runtime/src/impls.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Auxillary struct/enums for Statemint runtime.
+//! Taken from polkadot/runtime/common (at a21cd64) and adapted for Statemint.
+
+use crate::NegativeImbalance;
+use frame_support::traits::{Currency, Imbalance, OnUnbalanced};
+
+/// Logic for the author to get a portion of fees.
+pub struct ToAuthor<R>(sp_std::marker::PhantomData<R>);
+impl<R> OnUnbalanced<NegativeImbalance<R>> for ToAuthor<R>
+where
+	R: pallet_balances::Config + pallet_authorship::Config,
+	<R as frame_system::Config>::AccountId: From<polkadot_primitives::v1::AccountId>,
+	<R as frame_system::Config>::AccountId: Into<polkadot_primitives::v1::AccountId>,
+	<R as frame_system::Config>::Event: From<pallet_balances::Event<R>>,
+{
+	fn on_nonzero_unbalanced(amount: NegativeImbalance<R>) {
+		let numeric_amount = amount.peek();
+		let author = <pallet_authorship::Module<R>>::author();
+		<pallet_balances::Pallet<R>>::resolve_creating(
+			&<pallet_authorship::Module<R>>::author(),
+			amount,
+		);
+		<frame_system::Pallet<R>>::deposit_event(pallet_balances::Event::Deposit(
+			author,
+			numeric_amount,
+		));
+	}
+}
+
+pub struct DealWithFees<R>(sp_std::marker::PhantomData<R>);
+impl<R> OnUnbalanced<NegativeImbalance<R>> for DealWithFees<R>
+where
+	R: pallet_balances::Config + pallet_authorship::Config,
+	<R as frame_system::Config>::AccountId: From<polkadot_primitives::v1::AccountId>,
+	<R as frame_system::Config>::AccountId: Into<polkadot_primitives::v1::AccountId>,
+	<R as frame_system::Config>::Event: From<pallet_balances::Event<R>>,
+{
+	fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalance<R>>) {
+		if let Some(fees) = fees_then_tips.next() {
+			// for fees, 100% to author
+			// TODO: Figure out what to do with fees
+			let mut split = fees.ration(0, 100);
```

Comment:
I would assume this can all go, this is logic for the balance split.

```
if let Some(fees) = fees_then_tips.next() {
			// for fees, 100% to author
			// TODO: Figure out what to do with fees
			let mut split = fees.ration(0, 100);
			if let Some(tips) = fees_then_tips.next() {
				// for tips, if any, 100% to author
				tips.merge_into(&mut split.1);
			}
			<ToAuthor<R> as OnUnbalanced<_>>::on_unbalanced(split.1);
		}
```

can be this 

```
if let Some(fees) = fees_then_tips.next() {
			<ToAuthor<R> as OnUnbalanced<_>>::on_unbalanced(split.1);
		}
```

Reply:
that makes sense, then just more to dos and this PR is a thumbs up for me

Reply:
we could maybe aim to pass some of the fees to the treasury on the main chain